resources/js/components/QuizzesPanel/QuizComponent.vue

```diff
@@ -0,0 +1,227 @@
+quizRef.value.scheduledAt = formatDateHTML(quizRef.value.scheduledAt)
+const currentTime = ref(Date.now())
+const updateTimeInterval = setInterval(()=>currentTime.value = Date.now(), 1000)
+const confirmDeleteMessage = useMessageBox()
+onBeforeUnmount(()=>{clearInterval(updateTimeInterval)})
+
+const isReadyToSchedule = computed(()=>{
+  if (quizRef.value.scheduledAt && quizRef.value.duration)
+    return Date.parse(quizRef.value.scheduledAt) > currentTime.value
+  return false
+})
+
+// editing
+function edit(){
+  isEditing.value=true
+  if (!props.isSelected)
+    toggleQuizView()
+}
+function dismissEditing(){
+  quizRef.value = {...props.quiz}
+  isEditing.value=false
+}
+
+// date formatters
+function formatDatePretty(date?: DayjsConfigType) {
+  return date ? dayjs(date).format('DD.MM.YYYY HH:mm') : 'brak'
+}
+
+function formatDateHTML(date?: DayjsConfigType) {
+  return date ? dayjs(date).format('YYYY-MM-DDTHH:mm') : ''
+}
+
+function formatDateDB(date?: DayjsConfigType) {
+  return date ? dayjs(date).format('YYYY-MM-DD HH:mm:ss') : ''
+}
+
+function addQuestion(){
+  const newQuestion: CleanQuestion = { 
+    key: nanoid(),
+    text: 'Nowe pytanie',
+    answers: [],
+  }
+  quizRef.value.questions.push(newQuestion)
+}
+
+// quiz CRUD operations
+const request = new Request()
+
+function deleteQuiz() {
+  request.sendRequest(`/admin/quizzes/${quizRef.value.id}`, {method: 'delete'})
+}
+function copyQuiz() {
+  request.sendRequest(`/admin/quizzes/${quizRef.value.id}/clone`, {method: 'post'})
+}
+
+function assignDefinedValues<T extends object, U extends object>(target: T, source: U): T & U {
+  Object.keys(source).forEach((key) => {
+    const value = source[key as keyof U]
+    if (value) {
+      (target as any)[key] = value
+    }
+  })
+  return target as T & U
+}
+
+async function updateQuiz() {
+  let data : any = assignDefinedValues({}, quizRef.value)
+  if (data?.scheduledAt)
+    data.scheduledAt = formatDateDB(quizRef.value.scheduledAt)
+  data.onSuccess = ()=>isEditing.value=false 
+  await request.axiosPatch(`/admin/quizzes/${quizRef.value.id}`, data)
+}
+function schedule() {
+  if (!isReadyToSchedule.value) return
+  request.sendRequest(`/admin/quizzes/${quizRef.value.id}/lock`,{
+    method:'post',
+    onSuccess:()=>quizRef.value.state = 'locked',
+  })
+}
+
+function unSchedule() {
+  request.sendRequest(`/admin/quizzes/${quizRef.value.id}/unlock`,{
+    method:'post',
+    onSuccess:()=>quizRef.value.state = 'unlocked',
+  })
+}
+function deleteQuestion(question:CleanQuestion){
+  quizRef.value.questions = quizRef.value.questions.filter((q:CleanQuestion) => q.key !== question.key)
+}
+
+// emits
+function toggleQuizView() {
+  emit('displayToggle', quizRef.value)
+}
+
+// state
+const isPublished = computed(() => quizRef.value.state === 'published')
+const isDraft = computed(() => quizRef.value.state === 'unlocked')
+const isScheduled = computed(() => quizRef.value.state === 'locked')
+</script>
+
+<template>
```

Comment:
I suggest to create more components for seperate some code.

Reply:
Created new components for code separation